### PR TITLE
`PackedSha256` + `Sha256AndBlake2sInput ` hints

### LINKED ISSUE #540: `Sha256AndBlake2sInput` hint
`ids.full_word = int(ids.n_bytes >= 4)`

https://github.com/starkware-libs/cairo-lang/blob/0e4dab8a6065d80d1c726394f5d9d23cb451706a/src/starkware/starknet/security/whitelists/cairo_sha256.json#L118

https://github.com/starkware-libs/cairo-lang/blob/0e4dab8a6065d80d1c726394f5d9d23cb451706a/src/starkware/starknet/security/whitelists/cairo_blake2s.json#L129

https://github.com/lambdaclass/cairo-vm/blob/db7fff18c9c1312024ebf4119fcdfeba23657b12/cairo_programs/example_blake2s.cairo#L87C8-L87C45

https://github.com/lambdaclass/cairo-vm/blob/db7fff18c9c1312024ebf4119fcdfeba23657b12/vm/src/hint_processor/builtin_hint_processor/hint_code.rs#L803C11-L803C23

### LINKED ISSUE #567: `PackedSha256` hint
https://github.com/lambdaclass/cairo-vm/blob/db7fff18c9c1312024ebf4119fcdfeba23657b12/cairo_programs/packed_sha256.cairo#L335

https://github.com/starkware-libs/cairo-lang/blob/0e4dab8a6065d80d1c726394f5d9d23cb451706a/src/starkware/starknet/security/whitelists/cairo_sha256.json#L78

### PR DESCRIPTION
**Pull request by TAdev0**
Resolves #567 
Resolves #540

### CODE REVIEW

File: pkg/hintrunner/zero/zerohint_sha256.go [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+			output, err := hinter.ResolveAsAddress(vm, output)
+			if err != nil {
+				return err
+			}
+
+			for i := 0; i < len(newState); i++ {
+				newStateValue := mem.MemoryValueFromInt(newState[i])
+				memoryOffset := uint64(i)
+
+				err = vm.Memory.Write(output.SegmentIndex, output.Offset+memoryOffset, &newStateValue)
```
Comment (har777) [inline pkg/hintrunner/zero/zerohint_sha256.go:60]: safer to use `.AddOffset`

File: pkg/hintrunner/zero/zerohint_others.go [hunk 2]
```diff
@@ -751,3 +751,50 @@ func createNormalizeAddressHinter(resolver hintReferenceResolver) (hinter.Hinter
...
+
+// Sha256AndBlake2sInput hint writes 1 or 0 at `full_word` address, wether `n_bytes“
+// is greater than or equal to 4 or not
+//
+// `newSha256AndBlake2sInputHint` takes 2 arguments
+//   - `full_word` represents the address where the result of the comparison is stored
+//   - `n_bytes` represents the value that will be compared to 4
+func newSha256AndBlake2sInputHint(fullWord, nbytes hinter.ResOperander) hinter.Hinter {
+	return &GenericZeroHinter{
+		Name: "Sha256AndBlake2sInput",
```
Comment (har777) [inline pkg/hintrunner/zero/zerohint_others.go:763]: i would add a quick unit test for this too

File: pkg/utils/sha256.go [hunk 3]
```diff
@@ -0,0 +1,95 @@
+package utils
+
+func ComputeMessageSchedule(input []uint32) []uint32 {
+	// def compute_message_schedule(message: List[int]) -> List[int]:
+	// w = list(message)
+	// assert len(w) == 16
```
Comment (har777) [inline pkg/utils/sha256.go:6]: we should assert this, or change the type to enforce the length

File: pkg/utils/sha256.go [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+	//     w.append((w[i - 16] + s0 + w[i - 7] + s1) % 2**32)
+
+	// return w
+	w := make([]uint32, 64)
+	copy(w, input)
+
+	for i := 16; i < 64; i++ {
+		s0 := RightRot(w[i-15], 7) ^ RightRot(w[i-15], 18) ^ (w[i-15] >> 3)
+		s1 := RightRot(w[i-2], 17) ^ RightRot(w[i-2], 19) ^ (w[i-2] >> 10)
+		w[i] = w[i-16] + s0 + w[i-7] + s1
```
Comment (har777) [inline pkg/utils/sha256.go:20]: i see a `% 2**32` in the python code 🤔

File: pkg/utils/sha256.go [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+		0xa2bfe8a1, 0xa81a664b, 0xc24b8b70, 0xc76c51a3, 0xd192e819, 0xd6990624, 0xf40e3585, 0x106aa070,
+		0x19a4c116, 0x1e376c08, 0x2748774c, 0x34b0bcb5, 0x391c0cb3, 0x4ed8aa4a, 0x5b9cca4f, 0x682e6ff3,
+		0x748f82ee, 0x78a5636f, 0x84c87814, 0x8cc70208, 0x90befffa, 0xa4506ceb, 0xbef9a3f7, 0xc67178f2,
+	}
+
+	a, b, c, d, e, f, g, h := state[0], state[1], state[2], state[3], state[4], state[5], state[6], state[7]
+
+	for i := 0; i < 64; i++ {
+		S1 := RightRot(e, 6) ^ RightRot(e, 11) ^ RightRot(e, 25)
+		ch := (e & f) ^ ((^e) & g)
```
Comment (har777) [inline pkg/utils/sha256.go:75]: shouldnt it be `ch := (e & f) ^ ((~e) & g)`

File: pkg/utils/sha256.go [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+		temp1 := h + S1 + ch + k[i] + w[i]
```
Comment (har777) [inline pkg/utils/sha256.go:76]: shouldnt there be a `% 2**32` like the python code

File: pkg/utils/sha256.go [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+		S0 := RightRot(a, 2) ^ RightRot(a, 13) ^ RightRot(a, 22)
+		maj := (a & b) ^ (a & c) ^ (b & c)
+		temp2 := S0 + maj
```
Comment (har777) [inline pkg/utils/sha256.go:79]: shouldnt there be a % 2**32 like the python code

File: pkg/utils/sha256.go [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+
+		h = g
+		g = f
+		f = e
+		e = d + temp1
```
Comment (har777) [inline pkg/utils/sha256.go:84]: shouldnt there be a % 2**32 like the python code

File: pkg/utils/sha256.go [hunk 9]
```diff
@@ -0,0 +1,95 @@
...
+		d = c
+		c = b
+		b = a
+		a = temp1 + temp2
```
Comment (har777) [inline pkg/utils/sha256.go:88]: shouldnt there be a % 2**32 like the python code

File: pkg/utils/sha256.go [hunk 10]
```diff
@@ -0,0 +1,95 @@
...
+	}
+
+	return []uint32{
```
Comment (har777) [inline pkg/utils/sha256.go:91]: shouldnt there be a % 2**32 like the python code

File: pkg/hintrunner/zero/zerohint_others.go (hunk 2 above)
Comment (TAdev0) [reply pkg/hintrunner/zero/zerohint_others.go:763]: Sure

File: pkg/utils/sha256.go (hunk 3 above)
Comment (TAdev0) [reply pkg/utils/sha256.go:6]: Will add that check

Comment (TAdev0) [reply pkg/utils/sha256.go:8]: done!

File: pkg/utils/sha256.go (hunk 4 above)
Comment (har777) [reply pkg/utils/sha256.go:26]: uint32 wraps so closing these comments

File: pkg/hintrunner/zero/zerohint_others.go (hunk 2 above)
Comment (TAdev0) [reply pkg/hintrunner/zero/zerohint_others.go:763]: done!

File: pkg/hintrunner/zero/zerohint_sha256.go (hunk 1 above)
Comment (TAdev0) [reply pkg/hintrunner/zero/zerohint_sha256.go]: done!

File: pkg/utils/sha256.go (hunk 5 above)
Comment (TAdev0) [reply pkg/utils/sha256.go:81]: well, in Golang, `^e` means NOT, just like `~e` in python

Comment (TAdev0) [reply pkg/utils/sha256.go:81]: `^` can be NOT if applied to one operander, XOR if applied to 2 operators

Comment (har777) [reply pkg/utils/sha256.go:81]: aha!

**Review by har777 — APPROVED** [review]

**Pull request merged**